securedrop/journalist.py

```diff
@@ -394,12 +409,17 @@ def edit_account():
                            password=password)
 
 
-@app.route('/account/new-password', methods=['POST'])
+@app.route('/account/new-password', methods=('POST',))
 @login_required
 def new_password():
     user = g.user
-    password = request.form.get('password')
-    _set_diceware_password(user, password)
+    current_password = request.form.get('current_password')
+    token = request.form.get('token')
```

Comment:
I don't object to this, but I think we might not need the token for them to reset their account. This isn't the usual pattern on other webapps, but also SD is more paranoid. I'll like @redshiftzero to comment on this to see if they have an opinion.

Reply:
I would prefer the token should be present to change the user password.

Reply:
Yeah I hear you @heartsucker but I agree with @kushaldas - let's leave it in the PR as is with the token required. It's not much of an additional burden since they will need their 2FA token on hand anyway.